### ISSUE TITLE
[CLI] Implement train command wrapping DiffusionTrainer

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle#132]
## Background / Goal

The existing `unsloth_cli` train command wraps `UnslothTrainer`. For `unturtle_cli`, the train command must wrap `DiffusionTrainer` and expose dLLM-specific options (alpha scheduler, loss weight type, etc.) as CLI flags. No studio backend dependency — the command calls the unturtle Python API directly.

## Acceptance Criteria

- `unturtle_cli/commands/train.py` implements a `train` function
- CLI flags are auto-generated from `Config` via `@add_options_from_config(Config)`
- `--config/-c <path>` loads a YAML/JSON config file; CLI flags override config values
- `--dry-run` prints the resolved config as YAML and exits 0
- `unturtle train --model GSAI-ML/LLaDA-8B-Instruct --dataset <id> --dry-run` exits 0
- dLLM-specific flags present: `--alpha-scheduler`, `--loss-weight-type`, `--completion-only/--no-completion-only`, `--time-epsilon`
- `--hf-token` reads from `HF_TOKEN` env var if not provided
- `--wandb-token` reads from `WANDB_API_KEY` env var if not provided
- Training errors exit with code 1; argument errors with code 2

## Implementation Flow

```
1. @add_options_from_config(Config) → inject config_overrides
2. load config (--config file, then apply_overrides)
3. if --dry-run: print YAML, exit 0
4. validate: model and dataset required
5. FastDiffusionModel.from_pretrained(model, max_seq_length, load_in_4bit, token=hf_token)
6. if training_type == "lora": FastDiffusionModel.get_peft_model(model, r=lora_r, ...)
7. load dataset: datasets.load_dataset(dataset) or local JSONL/JSON
8. build DiffusionTrainingArguments from TrainingConfig + DiffusionConfig fields
9. DiffusionTrainer(model, args, train_dataset, processing_class=tokenizer)
10. trainer.train()
11. print final metrics
```

## Key Differences from unsloth_cli train

- Uses `FastDiffusionModel` not `FastLanguageModel`
- Uses `DiffusionTrainer` / `DiffusionTrainingArguments` not `UnslothTrainer`
- `processing_class=tokenizer` must be passed explicitly to `DiffusionTrainer`
- No background thread / studio integration needed

## Dependencies

- #129 (scaffold)
- #130 (options decorator)
- #131 (config model)

## Links

- `unturtle/diffusion/trainer.py` — `DiffusionTrainer`, `DiffusionTrainingArguments`
- `unturtle/fast_diffusion_model.py` — `FastDiffusionModel`
- Reference: `unsloth_cli/commands/train.py`
- Plan: `~/.claude/plans/hashed-tumbling-token.md` Issue D

**Comment by nishide-dev** [comment]
Closed by merged PR #143.
